spk/rutorrent/src/service-setup.sh

```diff
@@ -0,0 +1,289 @@
+# Usage: grant_basic_permissions "${SHARE_FOLDER}" "user:<user>"
+# Usage: grant_basic_permissions "${SHARE_FOLDER}" "group:<group>"
+grant_basic_permissions ()
+{
+    DIRNAME=`realpath "${1}"`
+    KEY="${2}"
+
+    VOLUME=$(echo "${DIRNAME}" | awk -F/ '{print "/"$2}')
+
+    # Ensure directory resides in /volumeX before setting GROUP permissions
+    if [ "`echo ${VOLUME} | cut -c2-7`" = "volume" ]; then
+        # Set read & execute permissions for KEY for folder and subfolders
+        if [ ! "`synoacltool -get \"${DIRNAME}\"| grep \"${KEY}:allow:r.x...a.R....:fd..\"`" ]; then
+            # First Unix permissions, but only if it's in Linux mode
+            if [ "`synoacltool -get \"${DIRNAME}\"| grep \"Linux mode\"`" ]; then
+                set_unix_permissions "${DIRNAME}"
+                # If it is linux mode (due to old package) we need to add "administrators"-group,
+                # otherwise the folder is not accessible from File Station anymore!
+                synoacltool -add "${DIRNAME}" "group:administrators:allow:rwxpdDaARWc:fd" >> ${INST_LOG} 2>&1
+            fi
+
+            # Then fix the Synology permissions
+            echo "Granting '${KEY}' basic permissions on ${DIRNAME}" >> ${INST_LOG}
+            synoacltool -add "${DIRNAME}" "${KEY}:allow:rxaR:fd" >> ${INST_LOG} 2>&1
+            find "${DIRNAME}" -type d -mindepth 1 -exec synoacltool -enforce-inherit "{}" \; >> ${INST_LOG} 2>&1
+        fi
+
+        # Walk up the tree and set traverse execute permissions for GROUP up to VOLUME
+        while [ "${DIRNAME}" != "${VOLUME}" ]; do
+            if [ ! "`synoacltool -get \"${DIRNAME}\"| grep \"${KEY}:allow:r.x...a.R\"`" ]; then
+                # Here we also need to make sure the admin can access data via File Station
+                if [ "`synoacltool -get \"${DIRNAME}\"| grep \"Linux mode\"`" ]; then
+                    synoacltool -add "${DIRNAME}" "group:administrators:allow:rwxpdDaARWc--:fd--" >> ${INST_LOG} 2>&1
+                fi
+                # Add the new group permissions
+                echo "Granting '${KEY}' basic permissions on ${DIRNAME}" >> ${INST_LOG}
+                synoacltool -add "${DIRNAME}" "${KEY}:allow:rxaR:n" >> ${INST_LOG} 2>&1
+            fi
+            DIRNAME="$(dirname "${DIRNAME}")"
+        done
+    else
+        echo "Skip granting '${KEY}' basic permissions on ${DIRNAME} as the directory does not reside in '/volumeX'. Set manually if needed." >> ${INST_LOG}
+    fi
+}
+
+service_preinst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        if [ ! -d "${wizard_download_dir}" ]; then
+            echo "Download directory ${wizard_download_dir} does not exist."
+            exit 1
+        fi
+        if [ -n "${wizard_watch_dir}" -a ! -d "${wizard_watch_dir}" ]; then
+            echo "Watch directory ${wizard_watch_dir} does not exist."
+            exit 1
+        fi
+    fi
+
+    exit 0
+}
+
+fix_shared_folders_rights()
```

Comment:
There are very specific folders on which rutorrent user *and* web user needs to be able to both read / write.